docs/auth/README.adoc

```diff
@@ -13,3 +13,128 @@ dependencies {
 
 == Usage
 
+To use the Auth SDK you'll first need to:
+
+* Import the Core module. See link:../core/README.adoc[the Core documentation].
+* Specify the authentication redirect URI in `android.defaultConfig.manifestPlaceholders` of the apps `build.gradle` file. See link:https://github.com/aerogear/aerogear-android-sdk/blob/master/example/build.gradle[example app build.gradle].
+* Specify the authentication redirect URI in the "valid redirect URIs" configuration of the Keycloak Client the app is authenticating with.
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+static int LOGIN_RESULT_CODE = 1;
+Activity myActivity = this.getActivity();
+
+// Initialise the auth service.
+AuthService authService = (AuthService) mobileCore.getInstance(AuthService.class);
+
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .allowSelfSignedCertificate(true)
+    .build();
+
+authService.init(context, authServiceConfig);
+
+// Build the credentials object and start the authentication flow.
+OIDCAuthenticateOptions credentials = new OIDCAuthenticateOptions(myActivity, LOGIN_RESULT_CODE);
+
+Callback authCallback = new Callback<Principal>() {
+    @Override
+    public void onSuccess(Principal principal) {
+        // User logged in, continue on..
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during login.
+    }
+}
+
+authService.login(credentials, authCallback);
+----
+
+Once the browser returns to the app the result can be handled. In the activity provided to the
+`login` method override `onActivityResult`. This handler should always invoke
+`AuthService#handleAuthResponse`, providing the `Intent`. This will exchange the temporary tokens
+returned from `AuthService#login` for long-life tokens and will provide a
+`UserPrincipal` which can be used to access a users details.
+
+More information about the user returned is available in link:../core/README.adoc[the auth module JavaDocs].
+
+[source,java]
+----
+@Override
+public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    if (requestCode == LOGIN_RESULT_CODE) {
+        // The core will return the same instance of the auth service as before
+        AuthService authService = mobileCore.getInstance(AuthService.class);
+        authService.handleAuthResponse(data);
+    }
+}
+----
+
+The callback provided in `AuthService#login` will be invoked.
+
+=== Retrieving the current user
```

Comment:
I say move this to the first and have the AuthService initialised here.  This is how most of the apps should use the module I think.

Reply:
Yeah that's a good point